backend/src/services/user/user-fns.ts

```diff
@@ -19,3 +21,54 @@ export const normalizeUsername = async (username: string, userDAL: Pick<TUserDAL
     }
   }
 };
+
+export const processFailedMfaAttempt = async (userId: string, userDAL: Pick<TUserDALFactory, "transaction">) => {
```

Comment:
Also i think as this logic is just used in the service layer you can keep it there itself.